x-pack/plugins/security_solution/public/common/components/user_privileges/endpoint/use_endpoint_privileges.ts

```diff
@@ -27,14 +28,20 @@ import { FleetAuthz } from '../../../../../../fleet/common';
  */
 export const useEndpointPrivileges = (): Immutable<EndpointPrivileges> => {
   const user = useCurrentUser();
-  const fleetServices = useKibana().services.fleet;
+  const fleetServicesFromUseKibana = useKibana().services.fleet;
+  // The `fleetServicesFromPluginStart` will be defined when this hooks called from a component
+  // that is being rendered under the Fleet context (UI extensions). The `fleetServicesFromUseKibana`
+  // above will be `undefined` in this case.
+  const fleetServicesFromPluginStart = useSecuritySolutionStartDependencies()?.fleet;
```

Comment:
@dasansol92 ,
Remember a few weeks ago we were looking at wanting to use `useEndpontPrivileges()` from inside of components that are rendered in fleet and we determined it was not possible? Well, in looking over the code again for this issue I'm fixing, I realized that we can grab the Fleet services we need from the Plugin's Start Dependencies instead of `useKibana()`. With this in place, we should now be able to use this Authz hook to conditionally show data depending on user's privileges.

I will open a separate issue to address hiding the cards when the user doe not have authz to endpoint. I also started to create a test renderer for UT, but did not include it in this PR because some refactoring is needed and I did not want to bloat this 8.2 backport.

Reply:
This is awesome, thanks for doing it!!